docs/log-levels.md

```diff
@@ -48,13 +49,13 @@ Some libraries and situations may not be entirely clear with regards to what log
 `trace` level logging:
 
 - Could include various additional information about a request, such as various diagnostics about created data structures, the state of caches or similar, which are created in order to serve a request.
-- Could include "begin operation" and "end operation" logs.
-  - However, please consider using [swift-distributed-tracing](https://github.com/apple/swift-distributed-tracing) to instrument "begin" and "end" events, as tracing can be more efficient than logging. Logging may need to create string representations of the durations, log levels, and other fields.
+- Could include "begin operation" and "end operation" logging statements.
+- However, please consider using [swift-distributed-tracing](https://github.com/apple/swift-distributed-tracing) to instrument "begin" and "end" events, as tracing can be more efficient than logging. Logging may need to create string representations of the durations, log levels, and other fields.
 
 `debug` level logging:
 
 - May include a single log statement for opening a connection, accepting a request, and so on.
-- It can include a _high level_ overview of what is going on in the library. For example: "started work, processing step X, finished work X, result code 200".
+- It can include a _high level_ overview of control flow in an operation. For example: "started work, processing step X, made X decision, finished work X, result code 200".  This overview may consist of high cardinality structured data.
```

Comment:
This is a vague statement.  Opening a connection can be a single log statement, but that implies that a high level overview is multiple statements.  If it's a single "event" then the overview should not be in text format, but should be a number of key/value pairs, but that also has drawbacks in that an event has a single timestamp attached to it, whereas the overview may want distinct timestamps for "started work", "processing step" etc.

Also I'd consider decision points such as handling special cases and if/else branches to be part of this as well.

Reply:
Those are good hints, thanks! I'll work something in